### PR TITLE
docs: Fix typo in deprecation message of `with_row_count`

### PR DESCRIPTION
**Pull request by langestefan**

### CODE REVIEW

File: py-polars/polars/dataframe/frame.py [hunk 1]
```diff
@@ -5306,7 +5306,7 @@ def with_row_count(self, name: str = "row_nr", offset: int = 0) -> Self:
         Add a column at index 0 that counts the rows.
 
         .. deprecated::
-            Use `meth`:with_row_index` instead.
+            Use `with_row_index` instead.
```
Comment (MarcoGorelli) [inline py-polars/polars/dataframe/frame.py:5309]: I think it should be
```
:meth:`with_row_index`
```
does it look alright if you make that change?

Comment (stinodego) [reply py-polars/polars/dataframe/frame.py:5309]: Yea that should be it.

**Review by stinodego — APPROVED** [review]
Thanks for cleaning up my mess 😄

**Comment by mcrumiller** [comment]
Can you also add a space in the first line of the `@deprecate_function` message (for both df and lazy)? It currently renders without a space after the period:
```
>>> pl.DataFrame().with_row_count()
...Use `with_row_index` instead.Note that ...
```

**Comment by stinodego** [comment]
> Can you also add a space

Done, thanks!

**Pull request merged**